app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -71,113 +71,132 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
-
-        $scope.emsCommonModel.service_account                 = data.service_account;
-        $scope.emsCommonModel.azure_tenant_id                 = data.azure_tenant_id;
-        $scope.emsCommonModel.keystone_v3_domain_id           = data.keystone_v3_domain_id;
-        $scope.emsCommonModel.subscription                    = data.subscription;
-
-        $scope.emsCommonModel.host_default_vnc_port_start     = data.host_default_vnc_port_start;
-        $scope.emsCommonModel.host_default_vnc_port_end       = data.host_default_vnc_port_end;
-
-        $scope.emsCommonModel.event_stream_selection          = data.event_stream_selection;
-
-        $scope.emsCommonModel.bearer_token_exists             = data.bearer_token_exists;
-
-        $scope.emsCommonModel.ems_controller                  = data.ems_controller;
-        $scope.emsCommonModel.default_auth_status             = data.default_auth_status;
-        $scope.emsCommonModel.amqp_auth_status                = data.amqp_auth_status;
-        $scope.emsCommonModel.service_account_auth_status     = data.service_account_auth_status;
-        $scope.emsCommonModel.metrics_auth_status             = data.metrics_auth_status;
-        $scope.emsCommonModel.ssh_keypair_auth_status         = data.ssh_keypair_auth_status;
-        $scope.emsCommonModel.hawkular_auth_status            = data.hawkular_auth_status;
-
-        if ($scope.emsCommonModel.default_userid != '') {
-          $scope.emsCommonModel.default_password = $scope.emsCommonModel.default_verify = miqService.storedPasswordPlaceholder;
-        }
-        if ($scope.emsCommonModel.amqp_userid != '') {
-          $scope.emsCommonModel.amqp_password = $scope.emsCommonModel.amqp_verify = miqService.storedPasswordPlaceholder;
-        }
-        if ($scope.emsCommonModel.metrics_userid != '') {
-          $scope.emsCommonModel.metrics_password = $scope.emsCommonModel.metrics_verify = miqService.storedPasswordPlaceholder;
-        }
-        if ($scope.emsCommonModel.ssh_keypair_userid != '') {
-          $scope.emsCommonModel.ssh_keypair_password = miqService.storedPasswordPlaceholder;
-        }
-        if ($scope.emsCommonModel.bearer_token_exists) {
-          $scope.emsCommonModel.default_userid = "_";
-          $scope.emsCommonModel.default_password = $scope.emsCommonModel.default_verify = miqService.storedPasswordPlaceholder;
-        }
-
-        $scope.afterGet  = true;
-        $scope.modelCopy = angular.copy( $scope.emsCommonModel );
-
-        $scope.populatePostValidationModel();
-
-        miqService.sparkleOff();
-      });
+      $http.get($scope.formFieldsUrl + emsCommonFormId)
+        .then(getEmsFormIdDataComplete)
+        .catch(getEmsFormDataFailed);
     }
     $scope.actionUrl = $scope.newRecord ? $scope.createUrl : $scope.updateUrl;
     $scope.currentTab = "default";
+
+    function getEmsFormIdDataComplete(response, status, headers, config) {
```

Comment:
You're right. I'll definitely go with what the <a href=https://docs.angularjs.org/api/ng/service/$http#get>Angular docs</a> say.

My source was <a href=https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y082>this</a>, which could be outdated.